src/Util/ZipFiles.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Util;
+
+use Doctrine\Common\Collections\Collection;
+
+use League\Flysystem\FilesystemInterface;
+
+use ZipStream\Option\Archive;
+use ZipStream\ZipStream;
+
+use App\Entity\File;
+
+class ZipFiles
+{
+    /**
+     * Flysystem interface object for Datastore.
+     *
+     * @var FilesystemInterface
+     */
+    private $datastoreFlysystem;
+
+    /**
+     * ZipFiles constructor.
+     *
+     * @param FilesystemInterface $datastoreFlysystem Datastore flystystem instance.
+     */
+    public function __construct(FilesystemInterface $datastoreFlysystem)
+    {
+        $this->datastoreFlysystem = $datastoreFlysystem;
+    }
+
+    /**
+     * Creates a zip file from collection of files.
+     *
+     * @param Collection|File $files   Files that need to be zipped.
+     * @param string          $zipFile Filename of the zipfile.
+     *
+     * @throws \Exception When utility class can not open/write to zip file.
+     *
+     * @return void
+     */
+    public function createZipFile(Collection $files, string $zipFile) : void
```

Comment:
I don't like giving this function the files collection, it has to know to do deal with files, and not be used for other things.
I think we should refactor this. Create another function called `addFiles` and just extent the `addFilesFromStream` function. You can create a new Archive and zipStream in the construction. Then finally create a `finish` function with the zip finish.